### PR TITLE
Change `MOVE` to seat and `owner`

### LINKED ISSUE #1116: Interaction of childrenPerOwner and MOVE to seat/hand
When working on a room, I ran into something that seems to be a bug.  If you use MOVE to seat (which then moves to hand), everything ultimately works out the way it is supposed to, but some intermediate step along the way is off.  In my room, I was trying to capture in a custom property the first player to have a card moved to his hand.  I tried doing this by having an enterRoutine that applied a custom property of the owner.  That didn't work so I tried putting a changeRoutine on the card so the first time owner was changed from null, it set a property onto the card.  That didn't work either.

What happened was the owner property got set correctly, but the custom property was set to whichever player pressed the button, so there was a mismatch.  I eventually got it to work by having everything happen in the same button that did the MOVE.  @robartsd summarized what he thinks is happening this way: "I think move to seat is moving to that hand (and childrenPerOwner in the hand is setting the owner to the player running the automation) then the seat is changing the owner to the player in the seat."  That's better than any summary I could come up with.  

There is a sort of demo at https://virtualtabletop.io/m9it.

### PR DESCRIPTION
**Pull request by RaphaelAlvez**

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: https://test.virtualtabletop.io/PR-1256/pr-test (or any other room on that server)

File: client/js/widgets/widget.js
Comment (ArnoldSmith86) [inline client/js/widgets/widget.js:1151]: That `else` belongs to the other `if` so you completely changed it here.

**Comment by 96LawDawg** [comment]
This breaks my heart.  I mean, it breaks the game Hearts.

**Comment by 96LawDawg** [comment]
I tested a bunch of other games using seats and owners, including several that would seem to me to be the most likely to have issues.  No problems now.  I guess this should be approved, but will wait on another code review from @ArnoldSmith86.

**Comment by RaphaelAlvez** [comment]
Great!!

@ArnoldSmith86 I think it was that last thing. (bringToFront thing). Basically because ether bug didn't look like the one I had that made me add the updateAfterShuffle.

But I think that there is no reason to just remove more code then necessary and it is working

**Comment by ArnoldSmith86** [comment]
The `c.bringToFront()` should really not be needed.

The other stuff is needed because `onChildAddAlign` calls `receiveCard` which in turn calls `rearrangeChildren` **but** only with `childrenOwned`. So it only rearranges your own and public cards because prior to seats that's probably all that could happen.

`updateAfterShuffle` calls `rearrangeChildren` with all cards.

In theory it doesn't have to be called after each card though. Once per target holder after the complete action should be enough.

That being said, just keep it the way it is. I just wanted to understand why removing it didn't work.

**Comment by RaphaelAlvez** [comment]
@96LawDawg 
> I tested a bunch of other games using seats and owners, including several that would seem to me to be the most likely to have issues. No problems now. I guess this should be approved, but will wait on another code review from @ArnoldSmith86.

@ArnoldSmith86 
> That being said, just keep it the way it is. I just wanted to understand why removing it didn't work.

review please

**Review by 96LawDawg — APPROVED** [review]
Approved based on testing.

**Review by ArnoldSmith86 — APPROVED** [review]
Code looks fine.

**Pull request merged**
